src/pip/_internal/download.py

```diff
@@ -722,7 +723,8 @@ def is_archive_file(name):
 
 def unpack_vcs_link(link, location):
     vcs_backend = _get_used_vcs_backend(link)
-    vcs_backend.unpack(location, url=link.url)
+    url = hide_url(link.url)
```

Comment:
Even though this seems harmless, I think it would be better to keep things narrow for now and just keep to the minimal places in the vcs code. We can expand the use in later PR's as the pattern solidifies, etc.